tool/planet/agent.go

```diff
@@ -481,6 +507,54 @@ func status(c statusConfig) (ok bool, err error) {
 	return ok, nil
 }
 
+// validateKubernetesService ensures that API server service has the IP from the
+// currently active service CIDR. In case of this not being the case (eg during service CIDR updates),
+// the function will remove the service and wait for Kubernetes API server to recreate it.
+func validateKubernetesService(ctx context.Context, serviceCIDR net.IPNet) error {
+	logger := log.WithField("service-cidr", serviceCIDR.String())
+	return utils.RetryWithInterval(ctx, newUnlimitedExponentialBackoff(5*time.Second), func() error {
+		client, err := cmd.GetKubeClientFromPath(constants.SchedulerConfigPath)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		services := client.CoreV1().Services(metav1.NamespaceDefault)
+		svc, err := services.Get(KubernetesServiceName, metav1.GetOptions{})
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		if svc.Spec.ClusterIP == "" {
+			// not a clusterIP service?
+			logger.Info("API server service clusterIP is empty.")
+			return nil
+		}
+		clusterIP := net.ParseIP(svc.Spec.ClusterIP)
+		// Kubernetes defaults to the first IP in service range for the API server service
+		// See https://github.com/kubernetes/kubernetes/blob/v1.15.12/pkg/master/services.go#L41
```

Comment:
Should this link point to K8s 1.17 sources?

Reply:
Done